proof/invariant-abstract/ARM/ArchDetSchedAux_AI.thy

```diff
@@ -16,27 +16,91 @@ lemmas arch_machine_ops_valid_sched_pred[wp] =
   arch_machine_ops_last_machine_time[THEN dmo_valid_sched_pred]
   arch_machine_ops_last_machine_time[THEN dmo_valid_sched_pred']
 
+lemma set_pt_eps_of[wp]:
+  "set_pt ptr pt \<lbrace>\<lambda>s. P (eps_of s)\<rbrace>"
+  by (set_object_easy_cases def: set_pt_def)
+
+lemma set_pd_eps_of[wp]:
+  "set_pd ptr pd \<lbrace>\<lambda>s. P (eps_of s)\<rbrace>"
+  by (set_object_easy_cases def: set_pd_def)
+
+lemma set_pt_ntfns_of[wp]:
+  "set_pt ptr pt \<lbrace>\<lambda>s. P (ntfns_of s)\<rbrace>"
+  by (set_object_easy_cases def: set_pt_def)
+
+lemma set_pd_ntfns_of[wp]:
+  "set_pd ptr pd \<lbrace>\<lambda>s. P (ntfns_of s)\<rbrace>"
+  by (set_object_easy_cases def: set_pd_def)
+
+lemma set_pt_prios_of[wp]:
+  "set_pt ptr pt \<lbrace>\<lambda>s. P (prios_of s)\<rbrace>"
+  by (set_object_easy_cases def: set_pt_def)
+
+lemma set_pd_prios_of[wp]:
+  "set_pd ptr pd \<lbrace>\<lambda>s. P (prios_of s)\<rbrace>"
+  by (set_object_easy_cases def: set_pd_def)
```

Comment:
We should be proving `P (tcbs_of s)` here (and wherever we can), because that's the more general case that includes `prios_of` and all other tcb projections.